src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -901,6 +917,68 @@ function discoverController(
     $scope.minimumVisibleRows = $scope.hits;
   };
 
+  $scope.onQuerySaved = savedQuery => {
+    $scope.savedQuery = savedQuery;
+  };
+
+  $scope.onSavedQueryUpdated = savedQuery => {
+    $scope.savedQuery = savedQuery;
+  };
+
+  $scope.onClearSavedQuery = () => {
+    delete $scope.savedQuery;
+    delete $state.savedQuery;
+    $state.query = {
+      query: '',
+      language: localStorage.get('kibana.userQueryLanguage') || config.get('search:queryLanguage'),
+    };
+    queryFilter.setFilters([]);
+    $state.save();
+    $scope.fetch();
+  };
+
+  const updateStateFromSavedQuery = (savedQuery) => {
+    $state.query = savedQuery.attributes.query;
```

Comment:
// brain-pick
Updating filter manager (and very soon timefilter) are copied in each app and they can be done inside the data plugin. Do you think this is possible \ relatively easy to do at the moment? (If it's hard, we can do it later)

Because it would pave the way to just doing `data.savedQueryManager.load()` and having it populate itself.

Reply:
Given the limited time we have it sounds like something we should do in a separate PR.